src/include/mutex.h

```diff
@@ -83,7 +93,17 @@ struct WT_COMPILER_TYPE_ALIGN(WT_CACHE_LINE_ALIGNMENT) __wt_spinlock {
 struct WT_COMPILER_TYPE_ALIGN(WT_CACHE_LINE_ALIGNMENT) __wt_spinlock {
 	wt_mutex_t lock;
 
-	const char *name;		/* Statistics: mutex name */
+	const char *name;		/* Mutex name */
+
+	/*
+	 * We track acquisitions and time spent waiting for some locks. For
+	 * performance reasons and to make it possible to write generic code
+	 * that tracks statistics for different locks, we store the offset
+	 * of the statistics fields to be updated during lock acquisition.
+	 */
+	int16_t stat_count;		/* acquisitions */
```

Comment:
@agorrod, I looked at something like this too, but decided against it. The code I came up with wasn't any better or smaller and I'd rather not have a new kind of spinlock (with new init/destroy functions and so on), appear in WiredTiger.

And, since spinlock structures are cache-line aligned, I don't think there's any space saving in the change, I think the compilers are creating (at a minimum), 64B structures.